sql/core/src/test/scala/org/apache/spark/sql/sources/v2/AlterTableTests.scala

```diff
@@ -0,0 +1,850 @@
+          StructField("x", DoubleType))), LongType)))
+    }
+  }
+
+  test("AlterTable: drop nested column in map value") {
+    val t = s"${catalogAndNamespace}table_name"
+    withTable(t) {
+      sql(s"CREATE TABLE $t (id int, points map<string, struct<x: double, y: double>>) " +
+        s"USING $v2Format")
+      sql(s"ALTER TABLE $t DROP COLUMN points.value.y")
+
+      val table = getTableMetadata(t)
+
+      assert(table.name === fullTableName(t))
+      assert(table.schema === new StructType()
+        .add("id", IntegerType)
+        .add("points", MapType(StringType, StructType(Seq(
+          StructField("x", DoubleType))))))
+    }
+  }
+
+  test("AlterTable: drop nested column in array element") {
+    val t = s"${catalogAndNamespace}table_name"
+    withTable(t) {
+      sql(s"CREATE TABLE $t (id int, points array<struct<x: double, y: double>>) USING $v2Format")
+      sql(s"ALTER TABLE $t DROP COLUMN points.element.y")
+
+      val table = getTableMetadata(t)
+
+      assert(table.name === fullTableName(t))
+      assert(table.schema === new StructType()
+        .add("id", IntegerType)
+        .add("points", ArrayType(StructType(Seq(
+          StructField("x", DoubleType))))))
+    }
+  }
+
+  test("AlterTable: drop column must exist") {
+    val t = s"${catalogAndNamespace}table_name"
+    withTable(t) {
+      sql(s"CREATE TABLE $t (id int) USING $v2Format")
+
+      val exc = intercept[AnalysisException] {
+        sql(s"ALTER TABLE $t DROP COLUMN data")
+      }
+
+      assert(exc.getMessage.contains("data"))
+      assert(exc.getMessage.contains("missing field"))
+    }
+  }
+
+  test("AlterTable: nested drop column must exist") {
+    val t = s"${catalogAndNamespace}table_name"
+    withTable(t) {
+      sql(s"CREATE TABLE $t (id int) USING $v2Format")
+
+      val exc = intercept[AnalysisException] {
+        sql(s"ALTER TABLE $t DROP COLUMN point.x")
+      }
+
+      assert(exc.getMessage.contains("point.x"))
+      assert(exc.getMessage.contains("missing field"))
+    }
+  }
+
+  test("AlterTable: set location") {
+    val t = s"${catalogAndNamespace}table_name"
+    withTable(t) {
+      sql(s"CREATE TABLE $t (id int) USING $v2Format")
+      sql(s"ALTER TABLE $t SET LOCATION 's3://bucket/path'")
+
+      val table = getTableMetadata(t)
+
+      assert(table.name === fullTableName(t))
+      assert(table.properties ===
+        Map("provider" -> v2Format, "location" -> "s3://bucket/path").asJava)
+    }
+  }
+
+  test("AlterTable: set table property") {
+    val t = s"${catalogAndNamespace}table_name"
+    withTable(t) {
+      sql(s"CREATE TABLE $t (id int) USING $v2Format")
+      sql(s"ALTER TABLE $t SET TBLPROPERTIES ('test'='34')")
+
+      val table = getTableMetadata(t)
+
+      assert(table.name === fullTableName(t))
+      assert(table.properties === Map("provider" -> v2Format, "test" -> "34").asJava)
+    }
+  }
+
+  test("AlterTable: remove table property") {
```

Comment:
shall we test updating table property?

Reply:
I just realized that these are existing tests. Feel free to address these comments in followup.